feat(livePage): 전체적인 UI 클론 구현 완료

ki2183 opened the pull request:
제목: `design(livePage): 대략적인 UI 구현 완료 네비게이션 토글 및 채팅창 toggle 기능 구현`

## 🚀 반영 브랜치
`livePage` → `develop`

<br/>

## ✅ 작업 내용

- 대략적인 외형적 구조 얼추 구현
- 임시로 네비게이션 토글 기능 구현
- 추후 Navigation 상태와 연동 예정
- zustand 전역 상태 변수를 이용한 채팅창 toggle 구현
- 대부분의 SVG 아이콘을 추출하여 public 디렉토리에 저장

<br/>

## 🌠 이미지 첨부
![일반](https://github.com/user-attachments/assets/36f14dcf-987f-44c2-8dd7-7797cebc0546)
![채팅창](https://github.com/user-attachments/assets/c4b2b04b-7e05-40fc-99d2-59df3cbf5046)
![네비게이션](https://github.com/user-attachments/assets/6e9960d4-7c88-4519-b772-79d2f17298f6)


<br/>

## ✏️ 앞으로의 과제

- 비디오 컨트롤러 구현
- 기능별 모달 구현

<br/>

## 📌 이슈 링크

- [livePage #2]

ki2183 committed:
feat(livePage): 전체적인 UI 클론 구현 완료

- 대략적인 외형적 구조 얼추 구현
- 임시로 네비게이션 토글 기능 구현
- 추후 Navigation 상태와 연동 예정
- zustand 전역 상태 변수를 이용한 채팅창 toggle 구현
- 대부분의 SVG 아이콘을 추출하여 public 디렉토리에 저장

Resolves:#2

DonggyunKim00 commented on src/app/(route)/live/[room_id]/components/Chat/ChatWindow.client.tsx:
```diff
@@ -0,0 +1,39 @@
+                닉네임
+            </span>
+
+            <span style={{ wordBreak: "break-word", textAlign: "left" }} className="text-[#2e3033] text-left break-words leading-[20px]">
```
inline style을 혼용한 이유가 있을까요?!

ki2183 replied:
앗.. 이건 tailwind로 바로 못 찾아서 급하게 적용한건데 tailwind로 수정하겠습니다

DonggyunKim00 commented on src/app/(route)/live/[room_id]/components/Chat/Input.client.tsx:
```diff
@@ -0,0 +1,29 @@
+
+function ChatInput() {
+  return (
+    <div id="live-chatting-area" className="flex-none p-[10px_20px] relative z-[100]">
```
기준님께서 id 값을 사용하신 의도가 궁금해요!

id 값을 사용해서 나중에 global css에 추가하여 고유하게 사용하는 방법도 좋을것 같다는 생각도 드네요!

ki2183 replied:
id는 작업하고 콘솔로 다시 확인할 때 도움되서 달았습니다

DonggyunKim00 commented on src/app/(route)/live/[room_id]/components/LiveStreaming/LiveDetails/HashTag.client.tsx:
```diff
@@ -0,0 +1,33 @@
+    
+  return (
+    <ul id="관련태그" className="text-[#666] mt-[8px] overflow-hidden flex flex-wrap text-[11px] gap-[4px] leading-[19px]">
+        {tags?.length > 0 &&
+            tags.map((tag, idx) => (
+            <HashTagLi key={`${tag.tagname}-${idx}`} {...tag} />
+        ))}
```
tags가 빈배열일때 아무것도 렌더링 하지 않는다는걸 의도하신걸까요?!?!
그렇다면 아래 코드가 더 깔끔해보이는데 기준님 생각은 어떠세요?!

```suggestion
        {tags.map((tag, idx) => (
          <HashTagLi key={`${tag.tagname}-${idx}`} {...tag} />
        ))}
```

ki2183 replied:
배열로 받을 때 혹시 태그가 없을 수 있어서 위처럼 코드를 짰어요
만약에 태그가 무조건 존재한다면 아래 코드로 수정하겠습니당

DonggyunKim00 replied:
태그가 없더라도 어차피 빈배열이기때문에 아무것도 렌더링되지 않을것같습니다!

DonggyunKim00 commented on src/app/(route)/live/[room_id]/components/_utils/SVGIcon.server.tsx:
```diff
@@ -0,0 +1,80 @@
+import Bell from "@/../public/CzzIcon/bell.svg";
```
제가 `@public` 경로 Alias 설정 해놨습니다! 아래처럼 경로 바꾸면 더 괜찮을것 같네요 👍 

```suggestion
import Bell from "@public/CzzIcon/bell.svg";
```

ki2183 replied:
아하 
아래 경로로 수정 하겠습니다

DonggyunKim00 commented on src/app/(route)/live/[room_id]/components/_utils/SVGIcon.server.tsx:
해당 컴포넌트 공통 컴포넌트화 시키면 좋을 것 같아요!
좀 더 동적으로 할당되면서 타입까지 가져가는 방식으로 디벨롭하면 좋을 것 같은데.. 바로 생각이 나진 않네요..

가은님이랑 리안님께서도 해당 코드에 대해 더 좋은 방안이 있다면 제안해 주시면 좋을 것 같아요! 👍
@yungan9 @AhnRian

ki2183 replied:
다른분들 동의 하신다면 공동 컴포넌트로 폴더로 옮기고, 더 괜찮은 타입이나 동적 할당 방법을 찾는다면 이슈에 올리거나 스크럼 때 건의 하겠습니다

ki2183 committed:
feat(livePage): PR 피드백 수정

- inline Style -> tailwind 수정
- tag 컴포넌트  조건 수정
- @/../public -> @public 경로 수정
- SvgIcon.server.tsx파일 src/public/_components 디렉토리로 이동

Resloves:#2
